cannon/mipsevm/multithreaded/mips.go

```diff
@@ -0,0 +1,318 @@
+package multithreaded
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm"
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm/exec"
+)
+
+func (m *InstrumentedState) handleSyscall() error {
+	thread := m.state.getCurrentThread()
+
+	syscallNum, a0, a1, a2, a3 := exec.GetSyscallArgs(m.state.GetRegisters())
+	v0 := uint32(0)
+	v1 := uint32(0)
+
+	//fmt.Printf("syscall: %d\n", syscallNum)
+	switch syscallNum {
+	case exec.SysMmap:
+		var newHeap uint32
+		v0, v1, newHeap = exec.HandleSysMmap(a0, a1, m.state.Heap)
+		m.state.Heap = newHeap
+	case exec.SysBrk:
+		v0 = exec.BrkStart
+	case exec.SysClone: // clone
+		// a0 = flag bitmask, a1 = stack pointer
+		if (a0 & exec.ValidCloneFlagsBitmask) != a0 {
```

Comment:
this is the same as:
```suggestion
		if a0 != exec.ValidCloneFlagsBitmask {
```

which is what we want since we don't support any unmasked clone bits

Reply:
Ah, ok.  I was thinking we would allow the supported flags to be on or off.  But it sounds like we only want to support the exact flags used by the go runtime.